client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/Util.java

```diff
@@ -54,4 +79,22 @@ protected static RuntimeException rewrap(ExecutionException e) {
          return new TransportException(e.getCause(), null);
       }
    }
+
+   public static boolean checkTransactionSupport(String cacheName, OperationsFactory factory, Log log) {
+      PrepareTransactionOperation op = factory.newPrepareTransactionOperation(DUMMY_XID, true, Collections.emptyList(),
+            false, 60000);
+      try {
+         return op.execute().handle((integer, throwable) -> {
+            if (throwable != null) {
+               log.invalidTxServerConfig(cacheName, throwable);
+            }
+            return throwable == null;
+         }).get();
```

Comment:
Wait should not be forever, there should be a time boundary and if exceeded return `false`?

Reply:
It doesn't wait forever. A scheduled task is registered after `sock-timeout` milliseconds and it cancels the operation.